device/alpaka/include/traccc/alpaka/utils/vecmem_types.hpp

```diff
@@ -0,0 +1,103 @@
+/**
```

Comment:
Not absolutely necessary to do it in this PR, but this header should eventually be in the `src/` directory. Generally we'll have to avoid any public dependence on the Alpaka headers in this library. Just like how we avoid any public CUDA or SYCL dependence in those libraries. (Not counting some coding mishaps, that still appear in some places...)